File: gdscript/src/main/java/gdscript/psi/impl/GdDictPatternImpl.java
package gdscript.psi.impl;

import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.psi.util.PsiTreeUtil;
import gdscript.psi.GdDictPattern;
import gdscript.psi.GdKeyValuePattern;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class GdDictPatternImpl extends ASTWrapperPsiElement implements GdDictPattern {

  public GdDictPatternImpl(@NotNull ASTNode node) {
    super(node);
  }

  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitDictPattern(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

  @Override
  @NotNull
  public List<GdKeyValuePattern> getKeyValuePatternList() {
    return PsiTreeUtil.getChildrenOfTypeAsList(this, GdKeyValuePattern.class);
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdDictPatternImpl.kt
package gdscript.psi.impl

import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.util.PsiTreeUtil
import gdscript.psi.GdDictPattern
import gdscript.psi.GdKeyValuePattern
import gdscript.psi.GdVisitor

class GdDictPatternImpl(node: ASTNode) : ASTWrapperPsiElement(node), GdDictPattern {
    fun accept(visitor: GdVisitor) {
        visitor.visitDictPattern(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override val keyValuePatternList: List<GdKeyValuePattern>
        get() = PsiTreeUtil.getChildrenOfTypeAsList(this, GdKeyValuePattern::class.java)
}
